### PR TITLE
Broadcast test runs

### PR DESCRIPTION
**Pull request by ErikSchierboom**
- Add serialization for test run
- Broadcast test run

### CODE REVIEW

File: test/commands/iteration/test_run/process_test.rb [hunk 1]
```diff
@@ -79,6 +79,7 @@ class Iteration::TestRun::ProcessTest < ActiveSupport::TestCase
 
     IterationChannel.expects(:broadcast!).with(iteration)
     IterationsChannel.expects(:broadcast!).with(iteration.solution)
+    TestRunChannel.expects(:broadcast!).with { |test_run| test_run == iteration.test_runs.last }
```
Comment (ErikSchierboom) [inline test/commands/iteration/test_run/process_test.rb]: Is this the way in which I can verify an expected value _after_ it has run? I couldn't just use `.with(iteration.test_runs.last)` like with the other calls, as the actual test run is created in the `Iteration::TestRun::Process.` call.

File: test/channels/test_run_channel_test.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+require "test_helper"
+
+class TestRunChannelTest < ActionCable::Channel::TestCase
+  test ".broadcast! broadcasts test run" do
```
Comment (ErikSchierboom) [inline test/channels/test_run_channel_test.rb:4]: If this is okay, I can also add tests for the `IterationChannel` and `IterationsChannel` classes.

Comment (iHiD) [reply test/channels/test_run_channel_test.rb:4]: This looks good to me.

File: test/commands/iteration/test_run/process_test.rb (hunk 1 above)
Comment (iHiD) [reply test/commands/iteration/test_run/process_test.rb]: This is absolutely fine. You could also do 
```
TestRunChannel.expects(:broadcast!).with(kind_of(Iteration::TestRun))
```

As you're not validating that the test run is particularly the "correct" test run, the `kind_of` check is probably just as safe. I don't have a preference over either. You could firm up the test by asserting that  `iteration.test_runs.size == 1` too

File: test/channels/test_run_channel_test.rb (hunk 2 above)
Comment (ErikSchierboom) [reply test/channels/test_run_channel_test.rb:4]: I've added tests for the other two untested channels too.

**Review by iHiD — APPROVED** [review]
Great. Thanks 🙂 

Tis red though 😢

**Comment by ErikSchierboom** [comment]
Should become green again :)

**Pull request merged**